### ISSUE TITLE
A DatePicker is abnormally slow in Edge and IE

### ISSUE THREAD

**Issue by alexeikostevich** [telerik/kendo-angular#1096]
## I'm submitting a...

* Bug report 

## Current behavior

It takes almost 4 seconds to open a [DatePicker](https://www.telerik.com/kendo-angular-ui/components/dateinputs/datepicker/#toc-datepicker-overview) in Edge and 12 seconds in IE 11.
Occasionally, it is noticeable how a _DatePicker_ repeatedly takes and loses a focus before being expanded.

Scrolling throw months is almost not responsive in IE.

## Expected behavior

A _DatePicker_ should provide its functionality without significant delays.

## Minimal reproduction of the problem with instructions

The issue is not reproducible in [demos](https://www.telerik.com/kendo-angular-ui/components/dateinputs/datepicker/#toc-forms-support) but is reproducible in all our products.

For example, this template
```html
<kendo-datepicker
     name="dueDate"
     [(ngModel)]="task.dueDate">
</kendo-datepicker>
```
behaves in the following way
- [in IE11](https://www.screencast.com/t/xOVcCfq1BSj) (reproducible)
- [in Google Chrome](https://www.screencast.com/t/gMba8hQV4) (not reproducible)

## What is the motivation or use case for changing the behavior?
The issue prevents from delivering good user experience in Edge/IE.

## Environment

Package versions:

```
+-- @angular/animations@5.0.2
+-- @angular/common@5.0.2
+-- @angular/compiler@5.0.2
+-- @angular/core@5.0.2
+-- @angular/forms@5.0.2
+-- @angular/platform-browser-dynamic@5.0.2
+-- @angular/router@5.0.2
+-- @progress/kendo-angular-buttons@2.0.0
+-- @progress/kendo-angular-dateinputs@1.4.0
+-- @progress/kendo-angular-dialog@1.3.0
+-- @progress/kendo-angular-dropdowns@1.2.4
+-- @progress/kendo-angular-excel-export@1.0.5
+-- @progress/kendo-angular-grid@1.6.1
+-- @progress/kendo-angular-inputs@1.3.3
+-- @progress/kendo-angular-intl@1.3.0
+-- @progress/kendo-angular-l10n@1.0.5
+-- @progress/kendo-angular-layout@1.0.10
+-- @progress/kendo-angular-popup@1.3.0
+-- @progress/kendo-angular-upload@2.0.2
+-- @progress/kendo-data-query@1.1.1
+-- @progress/kendo-date-math@1.1.0
+-- @progress/kendo-drawing@1.4.0
+-- @progress/kendo-theme-default@2.44.0
```

Browser:

- IE 11
- Edge version 38

System:
- TypeScript version: 2.4.2
- Node version: 6.10.3 (npm 4.6.1)
- Platform:  Windows

The issue is not noticeable in Google Chrome or Firefox.

**Comment by ggkrustev** [comment]
@alexeikostevich, thanks for the detailed report. I noticed that you are using `v1.4.0`, which introduced many improvements regarding the performance of the component (it become almost twice faster). 

Could you double check if that actual version used in the test demo is at least `1.4.0`? If it is then could you share a bit more details what your products look like? Would it be possible to share a simple demo that mimics the layout of the problematic page and reproduces the issue?
It will be really hard for us to continue with the investigation when we cannot observe the issue locally.

**Comment by alexeikostevich** [comment]
@ggkrustev , thank you for looking at this!

I confirm that 1.4.0 is used in the test demo. It behaves slightly better than 1.3.x but still not usable in IE and Edge for some reason.

I understand that it can be extremely difficult to research the issue without a demo. It can take some time to mimic this behaviour but I will try to share a demo shortly.

**Comment by ggkrustev** [comment]
@alexeikostevich, something that is worth noting when it comes to perf improvements is this article.

http://blog.mgechev.com/2017/11/11/faster-angular-applications-onpush-change-detection-immutable-part-1/

My assumption is that the app change detection (executed when NgZone.run is called) is somewhat slow in your case. I suppose that you can try to apply some of the suggestion give in the article if you haven't done some far. This checklist is also worth noting

https://github.com/mgechev/angular-performance-checklist

**Comment by alexeikostevich** [comment]
Thank you for sharing, @ggkrustev ! This is definitely the way to improve the overall performance for us.

Sorry for the late response: I have just finished my research. Trying to reproduce the issue in a demo app, I found out the root cause.

It seems that there is nothing wrong with Kendo UI DatePicker. It behaves slightly slow in IE, but this is natural for IE. 

The performance issue starts degrading dramatically when I add a CSS library based on attribute selectors. Surprisingly, just having many attribute selectors in CSS regardless of their usages, kills rendering performance in IE: https://medium.com/@antipov.alexei/ie11-performance-bottleneck-de304569361d

This is especially visible in DatePicker because it renders a lot of HTML dynamically.

I apologize for submitting the issue. We can safely close it. Hopefully, this finding can be helpful for someone else.

Thank you for looking at this!
